src/cr/cube/matrix/measure.py

```diff
@@ -1326,6 +1356,351 @@ def blocks(self):
         return ZscoreSubtotals.blocks(self._weighted_cube_counts, self._dimensions)
 
 
+# === MARGINALS ===
+
+
+class _BaseSecondOrderMarginal(object):
+    """Base class for all second-order measure objects."""
```

Comment:
I'm not sure there is such thing as a "first-order" marginal, is there? Wouldn't `_BaseMarginal` say all that needs to be said?

Docstring is unedited copy from `_BaseSecondOrderMeasure`.